great_expectations/rule_based_profiler/rule_based_profiler.py

```diff
@@ -295,27 +343,250 @@ def run(
+        }
+        override_rules: Dict[str, Rule] = {
+            rule_name: self._init_one_rule(rule_name=rule_name, rule_config=rule_config)
+            for rule_name, rule_config in override_rule_configs.items()
+        }
+        effective_rules.update(override_rules)
 
-        if pretty_print:
-            print(f"\nRuleBasedProfiler class name: {self.name}")
+        return list(effective_rules.values())
 
-            if not self._variables:
-                print(
-                    'Your current RuleBasedProfiler configuration has an empty "variables" attribute. \
-                    Please ensure you populate it if you\'d like to reference values in your "rules" attribute.'
+    @staticmethod
+    def _reconcile_rule(
+        existing_rules: Dict[str, Rule], rule_name: str, rule_config: dict
+    ) -> Dict[str, Any]:
+        effective_rule_config: Dict[str, Any]
+        if rule_name in existing_rules:
+            rule: Rule = existing_rules[rule_name]
+            domain_builder_config: dict = rule_config.get("domain_builder", {})
+            effective_domain_builder_config: dict = (
+                RuleBasedProfiler._reconcile_domain_builder_config(
+                    rule=rule,
+                    domain_builder_config=domain_builder_config,
                 )
+            )
+            effective_parameter_builder_configs: Optional[
+                List[dict]
+            ] = RuleBasedProfiler._reconcile_parameter_builder_configs_for_rule(
+                rule=rule,
+                rule_config=rule_config,
+            )
+
+            effective_expectation_configuration_builder_configs: List[
+                dict
+            ] = RuleBasedProfiler._reconcile_expectation_configuration_builder_configs_for_rule(
+                rule=rule,
+                rule_config=rule_config,
+            )
+            effective_rule_config = {
+                "domain_builder": effective_domain_builder_config,
+                "parameter_builders": effective_parameter_builder_configs,
+                "expectation_configuration_builders": effective_expectation_configuration_builder_configs,
+            }
+        else:
+            effective_rule_config = rule_config
+
+        return effective_rule_config
+
+    @staticmethod
+    def _reconcile_domain_builder_config(
+        rule: Rule,
+        domain_builder_config: dict,
+    ) -> dict:
+        effective_domain_builder_config: dict = {}
+        batch_request: Optional[
+            Union[BatchRequest, RuntimeBatchRequest, dict]
+        ] = domain_builder_config.pop("batch_request", None)
+        if batch_request is None:
+            batch_request = get_batch_request_as_dict(
+                batch_request=rule.domain_builder.batch_request
+            )
+
+        effective_domain_builder_config["batch_request"] = batch_request
+
+        key: str
+        value: Any
+        current_value: Any
+        for key, value in domain_builder_config:
```

Comment:
.items()?

Reply:
@cdkini Great catch -- fixed in all occurrences of this error.  Thanks!